### PR TITLE
go collector: add default metrics acceptance tests; adding more context to HELP

### PR DESCRIPTION
**Pull request by bwplotka**
The context and details for help were possible thanks to @vesari research, thanks for that!

cc @vesari

### CODE REVIEW

File: prometheus/go_collector.go [hunk 1]
```diff
@@ -116,7 +117,7 @@ func goRuntimeMemStats() memStatsMetrics {
 		}, {
 			desc: NewDesc(
 				memstatNamespace("heap_objects"),
-				"Number of allocated objects.",
+				"Number of currently allocated objects.",
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply prometheus/go_collector.go:120]: Hm, so this is current number, so it does not include released objects. Did you mean to add more details on some `_total` metric here? Like `mallocs_total` or `alloc_bytes_total`? Feel free to suggest with 

image

the exact description for those, we could add that `both live and gc-ed` line if you want, good idea 💪🏽

File: prometheus/go_collector.go [hunk 2]
```diff
@@ -225,7 +226,7 @@ func newBaseGoCollector() baseGoCollector {
 			nil, nil),
 		gcDesc: NewDesc(
 			"go_gc_duration_seconds",
-			"A summary of the pause duration of garbage collection cycles.",
+			"A summary of the wall-time pause (stop-the-world) duration in garbage collection cycles.",
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply prometheus/go_collector.go:229]: thanks! Especially as name is bit misleading, but anyone can use pause metric from runtime

File: prometheus/go_collector.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (vesari) [reply prometheus/go_collector.go:120]: You're right, it should be for the counter version, `mallocs_total`.

[the remark this answers is not in the document]
Comment (bwplotka) [reply prometheus/go_collector.go:120]: Can you propose a description that you want? (:

File: prometheus/go_collector.go [hunk 3]
```diff
@@ -60,23 +60,24 @@ func goRuntimeMemStats() memStatsMetrics {
 		}, {
 			desc: NewDesc(
 				memstatNamespace("mallocs_total"),
-				"Total number of mallocs.",
+				// TODO(bwplotka): We could add go_memstats_heap_objects, probably useful for discovery. Let's gather more feedback, kind of waste of bytes for everybody for compatibility reason.
+				"Total number of heap objects allocated, semantically a counter version for go_memstats_heap_objects gauge.",
```
Comment (vesari) [inline prometheus/go_collector.go:64]:
```suggestion
				"Total number of heap objects allocated, both live and gc-ed. Semantically a counter version for go_memstats_heap_objects gauge.",
```

**Review by vesari — COMMENTED** [review]
Re-submitting previous comment as code suggestion

**Pull request merged**